model/account.js

```diff
@@ -0,0 +1,62 @@
+'use strict';
+
+const crypto = require('crypto');
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+const mongoose = require('mongoose');
+const httpErrors = require('http-errors');
+
+const accountSchema = mongoose.Schema({
+  username: {type: String, required: true, unique: true},
+  email: {type: String, required: true, unique: true},
+  passwordHash: {type: String, required: true},
+  tokenSeed: {type: String, required: true, unique: true},
+  created: {type: Date, default: () => new Date()},
+});
+
+// Instance methods
+accountSchema.methods.passwordVerify = function(password) {
+  return bcrypt.compare(password, this.passwordHash)
+    .then(correctPassword => {
+      if(!correctPassword)
+        throw httpErrors(401, '__AUTH_ERROR__ incorrect password');
+      return this;
+    });
+};
+
+accountSchema.methods.tokenCreate = function() {
+  this.tokenSeed = crypto.randomBytes(64).toString('hex');
+  return this.save()
+    .then(account => {
+      let options = {expiresIn: '7d'};
+      return jwt.sign({tokenSeed: account.tokenSeed}, process.env.CLOUD_SECRET, options);
```

Comment:
this CLOUD_SECRET should probably be named after you app, or simply name it SECRET

CLOUD often is used to refer to 3d party cloud services